README.md

```diff
@@ -1,2 +1,234 @@
+[npm-image]:http://img.shields.io/npm/v/did-ipid.svg
+[travis-url]:https://travis-ci.org/ipfs-shipyard/js-ipid
+[travis-image]:http://img.shields.io/travis/ipfs-shipyard/js-ipid/master.svg
+[codecov-url]:https://codecov.io/gh/ipfs-shipyard/js-ipid
+[codecov-image]:https://img.shields.io/codecov/c/github/ipfs-shipyard/js-ipid/master.svg
+[david-dm-url]:https://david-dm.org/ipfs-shipyard/js-ipid
+[david-dm-image]:https://img.shields.io/david/ipfs-shipyard/js-ipid.svg
+[david-dm-dev-url]:https://david-dm.org/ipfs-shipyard/js-ipid?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/ipfs-shipyard/js-ipid.svg
+
+The [IPID DID method](https://github.com/jonnycrunch/ipid) implementation in JavaScript
+
+
+## Installation
+
+```sh
+$ npm install did-ipid
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants.
+
+If you target older browsers please make sure to transpile accordingly. 
+
+
+## Usage
+
+
+```js
+import IPID from 'did-ipid';
+
+const ipid = await IPID(ipfs);
+
+const didDocument = await ipid.resolve('did:ipid:QmUTE4cxTxihntPEFqTprgbqyyS9YRaRcC8FXp6PACEjFG');
+//=> Resolves a DID and returns the respective DID Document.
+
+const didDocument = await ipid.create(pem, (document) => {
+	const publicKey = document.addPublicKey({
+    	type: 'RsaVerificationKey2018',
+        publicKeyHex: '02b97c30de767f084ce3080168ee293053ba33b235d7116a3263d29f1450936b71',
+    });
+    
+    const authentication = document.addAuthentication(publicKey.id);
+    
+    const service = document.addService({
+    	id: 'hub',
+    	type: 'HubService',
+    	serviceEndpoint: 'https://hub.example.com/',
+  	});
+});
+//=> Creates a new DID and the corresponding DID Document based on the provided private key pem.
+//=> The DID Document is published with the added publicKey, authentication and service.
+
+const didDocument = await ipid.update(pem, (document) => {
+	document.revokeService('hub');
+ 
+    document.addService({
+    	id: 'messages',
+    	type: 'MessagingService',
+    	serviceEndpoint: '"https://example.com/messages/8377464',
+  	});
+});
+//=> Updates a DID Document based on the DID associated to the provided private key pem.
+//=> The DID Document is published without the `hub` service but with a new one called `messages`. 
+
+```
+
+## API
+
+### IPID
+
+An IPFS node is required to create an  IPID instance. Please be sure to check [js-ipfs](https://github.com/ipfs/js-ipfs), the JavaScript implementation of the IPFS protocol, to learn how to create one.
+
+Notes:
+- Please make sure that the IPFS node is `ready` to use.
+- This package uses the Key Management provided by IPFS. So during the creation of the node a password must be defined, as an option, to encrypt/decrypt your keys. 
+
+#### resolve(did)
+
+Resolves a DID and provides the respective DID Document.
+
+Returns a Promise that resolves to the DID Document.
+
+##### did
+Type: `String`
```

Comment:
Nifty pick: can we add a new line before this and in all similar situations in this doc?